com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -627,13 +661,27 @@ static IEnumerable<string> CheckOutputTensorPresence(Model model, int memory)
             var tensorTester = new Dictionary<string, Func<BrainParameters, ActuatorComponent[], TensorShape?, int, int, string>>();
 
             // This will need to change a bit for hybrid action spaces.
-            if (isContinuous == ModelActionType.Continuous)
+            if (!model.outputs.Contains(TensorNames.ContinuousActionOutput) && !model.outputs.Contains(TensorNames.DiscreteActionOutput))
             {
-                tensorTester[TensorNames.ActionOutput] = CheckContinuousActionOutputShape;
+                if (modelActionType == ModelActionType.Continuous)
```

Comment:
I think a lot of the changes in BarracudaModelParamLoader.cs could be simplified with an some extension methods on `Model`; something like
```
bool HasContinuousOutputs()
int ContinuousOutputSize()
string ContinuousOutputName()
```
and similar for discrete. All of those should handle the "legacy" case transparently. Then this code would simplify to
```
if(model.HasContinuousOutputs) {
  tensorTester[model.ContinuousOutputName] = CheckContinuousActionOutputShape;
}
if(model.HadDiscreteOutputs) {
  tensorTester[model.DiscreteOutputName] = CheckDiscreteActionOutputShape;
}
```
It could also simplify some of the new logic at the start of CheckModel()